### PR TITLE
Support Disable/Enable/Reset partition in Helix bootstrap tool

### PR DESCRIPTION
**Pull request by jsjtzyy**
1. support disabling partition when decommissioning a replica. (This is supposed to be done before removing it from IdealState)
2. support enabling partition if it is previously disabled
3. support resetting partition if it is previously in ERROR state
4. introduced HelixAdminOperation in bootstrap tool to simplify some code logic
5. added option to remove admin configs (i.e. PartitionOverride map)

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 1]
```diff
@@ -707,22 +722,45 @@ private void addUpdateResources(String dcName, Map<String, Set<String>> partitio
                 .add(partitionName);
           }
         } else if (!instanceSetInStatic.equals(instanceSetInHelix)) {
-          // @formatter:off
+          // we change the IdealState only when the operation is meant to bootstrap cluster or indeed update IdealState
+          if (EnumSet.of(HelixAdminOperation.UpdateIdealState, HelixAdminOperation.BootstrapCluster)
+              .contains(helixAdminOperation)) {
+            // @formatter:off
           info(
```
Comment (xuhao417347761) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:729]: Indent for the entire info from line 729 to 732.

**Comment by jsjtzyy** [comment]
@Arun-LinkedIn @xuhao417347761 Thanks for reviewing this PR. Please hold off on merging if you have approved. I plan to add `EnablePartition` and `ResetPartition` features as well. Thanks in advance.

**Comment by jsjtzyy** [comment]
@xuhao417347761 @Arun-LinkedIn This PR is ready for review.

**Comment by jsjtzyy** [comment]
I will keep both of you updated after I test them out in Perf cluster.

**Comment by jsjtzyy** [comment]
@xuhao417347761 @Arun-LinkedIn I have verified these features in Perf cluster, feel free to review and leave comments.

**Comment by xuhao417347761** [comment]
Looks good to me!

**Pull request merged**